packages/repo-tools/src/generate-sponsors.mts

```diff
@@ -118,25 +118,19 @@ async function main(): Promise<void> {
     requestGraphql<CollectiveData>('collective'),
   ]);
 
-  const accountsById = account.orders.nodes.reduce<
-    Record<string, MemberAccount>
-  >((accumulator, account) => {
-    const name = account.fromAccount.name || account.fromAccount.id;
-    accumulator[name] = {
-      ...accumulator[name],
-      ...account.fromAccount,
-    };
-    return accumulator;
-  }, {});
-
-  const totalDonationsById = collective.members.nodes.reduce<
-    Record<string, number>
-  >((accumulator, member) => {
+  const accountsById = Object.fromEntries(
```

Comment:
I'm not sure this has the same behavior.... The previous implementation copies and modifies values corresponding to existing keys in the `accumulator`, rather than overwriting them completely, which is what `Object.entries` does. Maybe that matters, maybe it doesn't, but I figure we should probably preserve the behavior as faithfully as possible with this PR...
 Maybe this one is better as a straightforward `for` loop?
 
 ```ts
 ```

Reply:
(note that converting to a for loop also forces you to clarify some confusing variable name shadowing)

Reply:
Good catch on the variable name that was just plain wrong — should be `order`, not `member`, `account` or `accountNode`. (The TypeScript type is `OrderNode`)

There is guaranteed to be no functionality change because:
=> no properties of the `OrderNode` type are optional (i.e. all properties are always there)
=> (I confirmed that was correct, on the actual response data)
=> so `accountNode.fromAccount` always overwrites `accountsById[name]`
=> `Object.fromEntries` picks the last value if two entries have the same key